gobblin-iceberg/src/main/java/org/apache/gobblin/iceberg/writer/IcebergMetadataWriter.java

```diff
@@ -731,6 +871,34 @@ public void flush(String dbName, String tableName) throws IOException {
     }
   }
 
+  /**
+   * For a sorted collection of timestamps greater than an existitng watermark, check audit counts for completeness between
+   * a source and reference tier with a granularit if 1 hour
+   * If the audit count matches update the watermark to the timestamp
+   * @param table
+   * @param timestamps
+   * @param previousWatermark
+   * @return updated completion watermark
+   */
+  private long computeCompletenessWatermark(String table, Collection<Long> timestamps, long previousWatermark) {
+    log.info(String.format("Compute completion watermark for %s and timestamps %s with previous watermark %s", table, timestamps, previousWatermark));
+    long completionWatermark = previousWatermark;
+    try {
+      for(long timestamp : timestamps) {
```

Comment:
Can we sort the timestamp in decreasing order and check for the first timestamp for which the audit counts match? Would make the code simpler and easier to read.
Also: I think this should be done as a binary search to minimize calls to audit system.

Reply:
Had a discussion with @omalley and it made more sense to be conservative and move in ascending order in case GMIP is lagging by a few hours

Reply:
Don't understand. Why should we be conservative? The goal should be to minimize calls to audit system.